### PR TITLE
feat: 为html/json转化为rss提供通用支持

### PR DESCRIPTION
**Pull request by ttttmr**
## 该 PR 相关 Issue / Involved Issue

Close #12204

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/rsshub/transform/html/https%3A%2F%2Fwechat2rss.xlab.app%2Fposts%2Flist%2F/item=div[class='post-content']%20p%20a
/rsshub/transform/html/https%3A%2F%2Fsec.today%2Fpulses%2F/item=div[class='card-body']
/rsshub/transform/html/http%3A%2F%2Fhyp3rlinx.altervista.org%2F/item=table[border=%221%22]%20tr%20td%20a
/rsshub/transform/json/https%3A%2F%2Fapi.github.com%2Frepos%2Fginuerzh%2Fgost%2Freleases/title=Gost%20releases&itemTitle=tag_name&itemLink=html_url&itemDesc=body
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

增加对通用的html/json解析，可以实现不改动rsshub代码，简单支持一些rss需求，也可以认为是简单替代版feed43

后续可以考虑做自动构建规则，类似 https://github.com/damoeb/rss-proxy ，目前做了非常简单的实现，填好基础的item选择器就能适配不少站了

### CODE REVIEW

**Comment by ttttmr** [comment]
https://github.com/DIYgod/RSSHub/discussions/12204

**Comment by ttttmr** [comment]
把几十个feed43迁移到rsshub了，总体上来说还不错，不少规则都是重复的

不会前端简单学一下[CSS 选择器](https://developer.mozilla.org/zh-CN/docs/Web/CSS/CSS_Selectors)应该也能快速上手

File: docs/other.md [hunk 1]
```diff
@@ -541,6 +541,76 @@ type 为 all 时，category 参数不支持 cost 和 free
...
+| `title`         | 指定 RSS 的标题                                        | `string` | 从当前网页中取`<title>` |
+| `item`          | 通过 CSS 选择器查找 HTML 元素作为`item`元素            | `string` | html                    |
+| `itemTitle`     | 在`item`中通过 CSS 选择器查找 HTML 元素作为`title`元素 | `string` | `item`元素              |
+| `itemTitleAttr` | 获取`title`元素属性作为标题                            | `string` | 元素 text               |
+| `itemLink`      | 在`item`中通过 CSS 选择器查找 HTML 元素作为`link`元素  | `string` | `item`元素              |
+| `itemLinkAttr`  | 获取`link`元素属性作为链接                             | `string` | `href`                  |
+| `itemDesc`      | 在`item`中通过 CSS 选择器查找 HTML 元素作为`desc`元素  | `string` | `item`元素              |
+| `itemDescAttr`  | 获取`desc`元素属性作为描述                             | `string` | 元素 html               |
+
+<Route author="ttttmr" example="/proxy/html/https%3A%2F%2Fwechat2rss.xlab.app%2Fposts%2Flist%2F/item=div[class='post-content']%20p%20a" path="/html/:url/:routeParams" :paramsDesc="[' URL地址，需要经过URL编码', '转换规则']">
```
Comment (TonyRL) [inline docs/other.md:563]:
```suggestion
<Route author="ttttmr" example="/proxy/html/https%3A%2F%2Fwechat2rss.xlab.app%2Fposts%2Flist%2F/item=div[class='post-content']%20p%20a" path="/html/:url/:routeParams" :paramsDesc="['URL地址，需要经过URL编码', '转换规则']">
```

The value of `routeParams` should be `encodeURIComponent`ed in the exmaple properly.
What you did is only `encodeURI`.

File: lib/v2/proxy/html.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+
+module.exports = async (ctx) => {
+    const { url } = ctx.params;
```
Comment (TonyRL) [inline lib/v2/proxy/html.js:5]: Please start both feed with
```js
    if (!config.feature.allow_user_supply_unsafe_domain) {
        ctx.throw(403, `This RSS is disabled unless 'ALLOW_USER_SUPPLY_UNSAFE_DOMAIN' is set to 'true'.`);
    }
```

**Review by TonyRL — COMMENTED** [review]
Is it possible to use name other than `proxy`?
`proxy` is for https://github.com/DIYgod/RSSHub/discussions/6671

**Comment by ttttmr** [comment]
@TonyRL I fixed the review

> Is it possible to use name other than `proxy`? `proxy` is for #6671

This name was used because it works similarly to the rss-proxy project, maybe `transform` is a good idea.
However, I have already deployed and used this name, and I have to modify dozens of rss links in Inoreader (delete, re-add and classify), which is quite troublesome

**Review by TonyRL — COMMENTED** [review]
> maybe `transform` is a good idea.

`transform` sound like a good idea to me.

> However, I have already deployed and used this name, and I have to modify dozens of rss links in Inoreader (delete, re-add and classify), which is quite troublesome

You can always maintain your own fork of RSSHub and customise it to suit your needs.

**Comment by ttttmr** [comment]
@TonyRL I renamed it

**Pull request merged**